sympy/functions/special/bessel.py

```diff
@@ -223,18 +223,33 @@ def _eval_rewrite_as_jn(self, nu, z, **kwargs):
 
     def _eval_as_leading_term(self, x, logx=None, cdir=0):
         nu, z = self.args
-        arg = z.as_leading_term(x)
-        if x in arg.free_symbols:
+        try:
+            arg = z.as_leading_term(x)
+        except NotImplementedError:
+            return self
+        c, e = arg.as_coeff_exponent(x)
+
+        if e.is_positive:
             return arg**nu/(2**nu*gamma(nu + 1))
-        else:
-            return self.func(nu, z.subs(x, 0))
+        elif e.is_negative:
+            cdir = 1 if cdir == 0 else cdir
+            sign = c*cdir**e
+            if not sign.is_negative and sign.is_real:
```

Comment:
Would `not sign.is_negative` suffice? I don't think it should necessarily be real.

Reply:
Yes, even I don't think that it would be required .